### ISSUE TITLE
Optimistic Write Issue with Paginated Queries

### ISSUE THREAD

**Issue by bradleypj823** [aws-samples/aws-mobile-appsync-events-starter-ios#8]
If you use the limit parameter (for paging) with the ListEventsQuery, writing to the cache in a optimistic update handler causes this exception to be caught: 

AWSAppSync.JSONDecodingError.missingValue

Heres the code, starting at line 42 on AddEventViewController:

```
appSyncClient?.perform(mutation: addEventMutation, optimisticUpdate: { (transaction) in
            do {
                // Update our normalized local store immediately for a responsive UI.
              try transaction?.update(query: ListEventsQuery(limit:10)) { (data: inout ListEventsQuery.Data) in
                    data.listEvents?.items?.append(
                        ListEventsQuery.Data.ListEvent.Item(id: UUID().uuidString,
                                                            description: descriptionText,
                                                            name: nameText,
                                                            when: whenText,
                                                            where: whereText,
                                                            comments: nil))
                    
                }
            } catch {
                print("Error updating the cache with optimistic response.")
       }
```

I've made sure I'm using the limit parameter every time I fetch with a ListEventsQuery. 
Any help would be greatly appreciated!

**Comment by rohandubal** [comment]
Hello @bradleypj823,

Thanks for reporting this. I will investigate this and reply back to you.

Best,
Rohan

**Comment by rodrigosol** [comment]
Any update about this?

**Comment by ChandraGillellamudi** [comment]
Any update on this?? it throwing an error even if I am not using limit

**Comment by richienko** [comment]
I got the same problem...
4 months investigating? Not exactly the behavior one is expecting from paid product.

**Comment by wir349** [comment]
Has this been fixed in 2.6.20?

**Comment by rohandubal** [comment]
This issue is resolved with SDK Version `2.6.22`.
